contracts/stake/src/multitest/distribution.rs

```diff
@@ -1136,225 +1273,54 @@ fn apr_cw20() {
-        .with_unbonding_periods(vec![unbonding_period])
-        .with_initial_balances(vec![
-            (&members[0], bonds[0]),
-            (&members[1], bonds[1]),
-            (&members[2], bonds[2]),
-            (&members[3], 400u128),
-        ])
-        .with_admin("admin")
-        .with_native_balances("juno", vec![(&members[3], 400)])
-        .build();
-
-    suite
-        .create_distribution_flow(
-            "admin",
-            &members[0],
-            AssetInfo::Native("juno".to_string()),
-            vec![(unbonding_period, Decimal::one())],
-        )
-        .unwrap();
-
-    assert_eq!(suite.query_balance_staking_contract().unwrap(), 0);
-
-    suite
-        .delegate(&members[0], bonds[0], unbonding_period)
-        .unwrap();
-    suite
-        .delegate(&members[1], bonds[1], unbonding_period)
-        .unwrap();
-    suite
-        .delegate(&members[2], bonds[2], unbonding_period)
-        .unwrap();
-
-    assert_eq!(suite.query_balance_staking_contract().unwrap(), delegated);
-
-    let _resp = suite
-        .distribute_funds(&members[3], None, Some(juno(400)))
-        .unwrap();
-
-    // resp.assert_event(&distribution_event(&members[3], &denom, 400));
-
-    // assert that staking token balance is still the same
-    assert_eq!(suite.query_balance_staking_contract().unwrap(), delegated);
-    // assert that rewards are there
-    assert_eq!(
-        suite
-            .query_balance(suite.stake_contract().as_str(), "juno")
-            .unwrap(),
-        400,
-    );
-
-    assert_eq!(suite.query_balance(&members[0], "juno").unwrap(), 0);
-    assert_eq!(suite.query_balance(&members[1], "juno").unwrap(), 0);
-    assert_eq!(suite.query_balance(&members[2], "juno").unwrap(), 0);
-    assert_eq!(suite.query_balance(&members[3], "juno").unwrap(), 0);
-
-    assert_eq!(
-        suite.withdrawable_rewards(&members[0]).unwrap(),
-        vec![juno(50)]
-    );
-    assert_eq!(
-        suite.withdrawable_rewards(&members[1]).unwrap(),
-        vec![juno(100)]
-    );
-    assert_eq!(
-        suite.withdrawable_rewards(&members[2]).unwrap(),
-        vec![juno(250)]
-    );
-
-    assert_eq!(suite.distributed_funds().unwrap(), vec![juno(400)]);
-    assert_eq!(suite.undistributed_funds().unwrap(), vec![juno(0)]);
-
-    suite.withdraw_funds(&members[0], None, None).unwrap();
-    suite.withdraw_funds(&members[1], None, None).unwrap();
-    suite.withdraw_funds(&members[2], None, None).unwrap();
-
-    // assert_eq!(
-    //     suite
-    //         .query_balance_vesting_contract(suite.stake_contract().as_str())
-    //         .unwrap(),
-    //     0
-    // );
-    assert_eq!(suite.query_balance(&members[0], "juno").unwrap(), 50);
-    assert_eq!(suite.query_balance(&members[1], "juno").unwrap(), 100);
-    assert_eq!(suite.query_balance(&members[2], "juno").unwrap(), 250);
-    assert_eq!(suite.query_balance(&members[3], "juno").unwrap(), 0);
-}
-
-#[test]
-fn divisible_amount_distributed_twice() {
```

Comment:
when it comes to distribution.rs these are the tests that I did not bring back, because the Msg that we test is no longer included in the contract:
`funds_withdrawal_delegation()`, `rebond_works()` and `rebond_multiple_works()`

we no longer have delegate and rebond msg and we also talked about that in DMs